src/hazelcore/UHExp.re

```diff
@@ -30,6 +30,7 @@ and operand =
   | Inj(ErrStatus.t, InjSide.t, t)
   | Case(CaseErrStatus.t, t, rules)
   | Parenthesized(t)
+  | UnaryOp(ErrStatus.t, operator, operand)
```

Comment:
you should define a separate datatype for unary operators, rather than using `operator` (which should be renamed `binop` now)

Reply:
Ok got it